Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs

```diff
@@ -194,6 +215,10 @@ private void UpdateAppearance(EntityUid uid, AppearanceComponent? appearance = n
                 return;
 
             _appearance.SetData(uid, KitchenSpikeVisuals.Status, component.PrototypesToSpawn?.Count > 0 ? KitchenSpikeStatus.Bloody : KitchenSpikeStatus.Empty, appearance);
+
+            var proto = Prototype(uid);
+            if (component.PrototypesToSpawn?.Count == 0 && proto != null)
+                _metaData.SetEntityName(uid, proto.Name);
```

Comment:
what is this code even doing?

Reply:
This code resets the meat spike name. 
> and while spiked it displays the creature's name.
When the meat spike is finished providing meat (i.e. PrototypesToSpawn is 0), the meat spike is given the original prototype name.

Reply:
Prototypes are only to be used for spawning. If anything touches this (e.g. a labeller) it's just going to break.

Reply:
So I have looked further into this and my conclusion is that the underlying system is broken, not my specific implementation. There is no real support for "remembering" the name and any system which renames an object is incompatible with any other system.

As an example, the label system saves the pre-label name in the entity's LabelComponent. This is the name that the entity gets reset to when the label is removed. That means any other system that changes an object name gets overwritten if the label is removed:

- You can remove the pAI's name and return it to "personal ai device" by labeling it before a role joins, and then removing the label after.
- You can remove *glue and lube tags* from an object by labeling it before applying the substance and then unlabeling it after.
- Depending on implementation, this can happen in reverse.
  - For glue/lube, if you apply the substance to a labeled object and then remove the label, the object gets the label back when the effect wears off.
  - For pAIs (which uses the implementation I used), the label gets hidden when a pAI role joins and remains hidden once the role leaves as the entity gets reset to its prototype name.

Metamorphic glasses is another system that has this same problem.

I am going to post this as a separate issue. While I know that it's preferred that we don't add to broken systems but fix them instead, fixing the above properly would require enough work to be its own PR that likely needs to touch RobustToolbox code (of which I have no experience). Let me know how you think I should proceed.